### PR TITLE
Feat plants crud db

### PR DESCRIPTION
**Pull request by Xemko**
Adds basic CRUD to plants logic. Tests are ok.
Do we need room collection as well? or room field is enough in plant document? And you going to sort it by room on the front end?

### CODE REVIEW

File: apps/server/requests.rest [hunk 1]
```diff
@@ -26,11 +26,15 @@ POST http://localhost:3000/api/plants/
 Content-Type: application/json
 
 {
-    "name": "Cactus1",
-
-    "description": "Cactus is a plant",
-    "image": "placeholder",
+    "name": "notCactus",
```
Comment (tgcaspa) [inline apps/server/requests.rest:29]: iq = 500100

File: apps/server/src/controllers/auth.controller.js [hunk 2]
```diff
@@ -24,20 +23,16 @@ try {
 
 const signIn = async (req, res) => {
     const { phoneNumber } = req.body;
-    console.log(phoneNumber);
     if (!phoneNumber) {
        return res.status(401).json({code: 401, message: "Please enter a valid phone number"});
     }
     try {
         const user = await authService.findUserByPhoneNumber(phoneNumber);
-        if (!user) {
```
Comment (tgcaspa) [inline apps/server/src/controllers/auth.controller.js:33]: I think this if is more important than above. combine them together

File: apps/server/src/controllers/plant.controller.js [hunk 3]
```diff
@@ -36,8 +35,34 @@ try {
...
 }
 
+const updatePlantById = async (req, res) => {
+    try {
+        const plantId = req.params.plantId;
+        const plantData = req.body;
+
+        const updatedPlant = await plantService.updatePlantById(plantId, plantData);
+
+        return res.status(200).json({code: 200, message: 'Success', updatedPlant});
```
Comment (tgcaspa) [inline apps/server/src/controllers/plant.controller.js]: please return `plant` insteadof `updatedPlant`

Comment (tgcaspa) [reply apps/server/src/controllers/plant.controller.js]: should not break the interface

File: apps/server/src/controllers/plant.controller.js [hunk 4]
```diff
@@ -36,8 +35,34 @@ try {
...
+    } catch (err) {
+        return res.status(500).json({code: 500, message: 'Internal server error'});
+    }
+}
+
+const deletePlantById = async (req, res) => {
+    try {
+        const plantId = req.params.plantId;
+        const deletedPlant = await plantService.deletePlantById(plantId);
+        return res.status(200).json({code: 200, message: 'Success', deletedPlant});
```
Comment (tgcaspa) [inline apps/server/src/controllers/plant.controller.js]: the same -> `plant`

Comment (Xemko) [reply apps/server/src/controllers/plant.controller.js]: Done

File: apps/server/src/controllers/plant.controller.js [hunk 5]
```diff
@@ -2,18 +2,18 @@ const plantService = require('../services/plant.service');
...
     } catch (err) {
         return res.status(500).json({code: 500, message: 'Internal server error'});
     }
 }
 
 const getPlantsByUserId = async (req, res) => {
     try {
-        const userId = req.params.userId;
+        const userId = req.user.id;
+        console.log(userId);
```
Comment (tgcaspa) [inline apps/server/src/controllers/plant.controller.js:16]: W: zaebal

File: apps/server/src/services/plant.service.js [hunk 6]
```diff
@@ -7,14 +7,26 @@ const createPlant = async (plantData) => {
...
-    return await Plant.find({userId : userId});
+    const plants = await Plant.find({userId});
+    return plants;
+}
+const updatePlantById = async (plantId, plantData) => {
+    const updatedPlant = await Plant.findByIdAndUpdate(plantId, plantData, {new: true});
+    return updatedPlant;
 }
+const deletePlantById = async (plantId) => {
+    const deletedPlant = await Plant.findByIdAndDelete(plantId);
```
Comment (tgcaspa) [inline apps/server/src/services/plant.service.js:22]: C: return it directly

**Review by tgcaspa — APPROVED** [review]

File: apps/server/src/controllers/plant.controller.js (hunk 5 above)
Comment (Xemko) [reply apps/server/src/controllers/plant.controller.js:16]: Tak debuging jej delal.

File: apps/server/src/services/plant.service.js (hunk 6 above)
Comment (Xemko) [reply apps/server/src/services/plant.service.js:22]: Return await plant? 💯

**Pull request merged**